gateway/enforcer/org.wso2.apk.enforcer/src/main/java/org/wso2/apk/enforcer/security/AuthFilter.java

```diff
@@ -166,6 +166,7 @@ public boolean handleRequest(RequestContext requestContext) {
                 if (authenticateResponse.isMandatoryAuthentication()) {
                     authenticated = authenticateResponse.isAuthenticated();
                     setInterceptorAuthContextMetadata(authenticator, requestContext);
+                    setInterceptorAPIMetadata(requestContext);
```

Comment:
why do we only set this under authenticateResponse.isMandatoryAuthentication() condition only?

Reply:
moved it to `handleRequest` method body level without enclosed conditions